### DISCUSSION TITLE
Parseq Compatible input sizes

### DISCUSSION THREAD

**Discussion by bowentkruse** [mindee/doctr#1413] [Q&A]
Is there a set of input sizes that parseq models are limited to?

In the past I've been able to train models with other architectures on a custom dataset with different values for the --input_size value with the only limitation being the memory of the devices training the model. Now however, while attempting to change the --input_size value, both smaller and larger than the default of 32, I experience an error. 

**Call of train_pyorch.py:**
`python train_pytorch.py parseq --train_path=stencils/train --val_path=stencils/validate --vocab english --epochs 550 --lr 0.00007 --min-chars=4 --max-chars=6 --pretrained -b 8 --wd 0.001 --input_size 40`

**Error produced:**
```
File "/pathtomyproject/doctr/models/modules/vision_transformer/pytorch.py", line 84, in forward
    embeddings += self.positions
RuntimeError: The size of tensor a (201) must match the size of tensor b (129) at non-singleton dimension 1

```

My level of expertise on the implementation of things like weight decay and min/max chars is very limited so I tried changing the input_size variable with and without changing those variables and the issue persisted.

**Comment by bowentkruse** [comment]
@felixdittrich92 Is this also possibly a bug? If so I'll open bug report.

**Reply by felixdittrich92** [reply]
Hi @bowenkruse :wave:,

sry for the late response :)
In this case it's not a bug because the ViT backbone we use does not support dynamic generation of patches which leads to this weight mismatch.
What is the reason that you want to use larger image crops for training?

**Reply by bowentkruse** [reply]
Hello @felixdittrich92, while playing around with the SAR architecture I noticed I was able to get the validation and testing accuracy of models much higher by tuning input size. My assumption (which very well could be wrong) is that I can get similar boosts in accuracy by tuning input size for the PARSeq architecture as well.
